tso-server/src/main/java/com/yahoo/omid/tso/RetryProcessorImpl.java

```diff
@@ -99,24 +109,30 @@ public void onEvent(final RetryEvent event, final long sequence, final boolean e
 
     }
 
-    private void handleCommitRetry(RetryEvent event) throws InterruptedException, ExecutionException {
+    public void addAbort(Batch batch, long startTimestamp, Channel c, MonitoringContext context) {
+        batch.addAbort(startTimestamp, true, c, context);
+    }
 
-        long startTimestamp = event.getStartTimestamp();
+    public void addCommit(Batch batch, long startTimestamp, long commitTimestamp, Channel c, MonitoringContext context) {
+        batch.addCommit(startTimestamp, commitTimestamp, c, context);
+    }
 
+    private void handleCommitRetry(RetryEvent event) throws InterruptedException, ExecutionException {
+        long startTimestamp = event.getStartTimestamp();
         try {
             Optional<CommitTimestamp> commitTimestamp = commitTableClient.getCommitTimestamp(startTimestamp).get();
-            if (commitTimestamp.isPresent()) {
+            Batch batch = new Batch(1, -1);
```

Comment:
(-1) denotes that this batch should be immediately processed in the reply processor without any order. I replaced it with the appropriate macro.